### PR TITLE
Add cursor type for more safety.

### PR DESCRIPTION
**Pull request by nickmshelley**

### CODE REVIEW

File: Sources/SQLite/Typed/Query.swift [hunk 1]
```diff
@@ -894,58 +894,89 @@ public struct Delete : ExpressionType {
...
+                            try expandGlob(true)(q)
+                            continue column
                         }
                         throw QueryError.noSuchTable(name: namespace)
                     }
-                    for q in queries {
-                        try expandGlob(query.clauses.join.count > 0)(q)
-                    }
-                    continue
+                    fatalError("no such table: \(namespace)")
```
Comment (jberkel) [inline Sources/SQLite/Typed/Query.swift]: this should probably be `QueryError.noSuchTable`?

Comment (nickmshelley) [reply Sources/SQLite/Typed/Query.swift]: Good call. Sorry I took so long to make this simple change.

**Comment by jberkel** [comment]
I merged this into the release branch, with some additional changes (5a9b583d2a28d12bc8523b204e75a0da6c8b04ff). I'm wondering if `RowCursor` will be confused with `Cursor` on statement, maybe a different name would be more appropriate? `RowIterator`?

**Comment by nickmshelley** [comment]
Yeah `RowIterator` sounds fine to me. Thanks for getting it in!

**Pull request closed without merging**